exercises/run-length-encoding/example.rs

```diff
@@ -0,0 +1,45 @@
+use std::cmp;
+
+pub fn encode(input: &str) -> String {
+    input
+        .chars()
+        .fold((String::new(),' ',0,1), |(mut acc, last, last_n, pos), c| {
+            // acc = where answer is accumulated
+            // last = last character read
+            // last_n = accum count for last
+            if c == last {
+                if pos == input.len() { // end of string
+                    acc += (last_n + 1).to_string().as_str();
+                    acc.push(c);
+                }
+                (acc, last, last_n + 1, pos + 1)
+            }
+            else {
+                if last_n > 1 {
+                    acc += last_n.to_string().as_str();
+                }
+                if last_n > 0 { // ignore initial last (single whitespace)
+                    acc.push(last);
+                }
+                if pos == input.len() { // end of string
+                    acc.push(c);
+                }
+                (acc, c, 1, pos + 1)
+            }
+        }).0
+}
+
+pub fn decode(input: &str) -> String {
+    input
+        .chars()
+        .fold((String::new(), 0), |(mut acc, last_n), c| {
+            if c.is_digit(10) { // keep reading.. keeeep reading
```

Comment:
haha sorry, the idea is that you can't conclude that the number of digits to place is c because there may be other digits latter. This is just a personal comment, i'll remove it